### ISSUE TITLE
Transparency doesn't work when background colour is not the default

### ISSUE THREAD

**Issue by bzzrak** [microsoft/terminal#15345]
### Windows Terminal version

1.17.1023

### Windows build number

10.0.22621.1702

### Other Software

cmd.exe, vim...

### Steps to reproduce

Run `color f0` in cmd.exe, or for example `colo peachpuff` in Vim. Anything that changes the background colour

### Expected Behavior

Transparency works like in Conhost
<img width="300" alt="image" src="https://github.com/microsoft/terminal/assets/77467869/be0b3df8-6e4b-4c56-ae77-02e5b1997b8d">

### Actual Behavior

Transparency doesn't work, except on the right edge of the window
<img width="300" alt="image" src="https://github.com/microsoft/terminal/assets/77467869/2d3eb223-b86d-471e-8cf0-257062ab1028">

**Comment by j4james** [comment]
If I've understood you correctly, I think the feature you want is tracked in #5145.

**Comment by bzzrak** [comment]
That doesn't seem like what I'm talking about. That issue is about blending an image background with the default background colour. 
I'm talking about the fact that transparency just doesn't work when the background colour isn't the default

**Comment by j4james** [comment]
I think the underlying issue is the same though - the way the transparency is currently implemented, it only applies to the default background color. In your case that means you can't see the desktop. In their case, they can't see their background image. Both case would likely be addressed in the same way.

I might be wrong, though. One the core devs will be able to confirm one way or the other.

**Comment by lhecker** [comment]
Yep, I agree. This is fundamentally the same issue, given the way we've implemented background colors. Thanks for reporting this however! If you can, please upvote the other issue. We do use the number of upvotes on issues to determine what features to implement next from time to time.

/dup #5145
